components/eamxx/scripts/cime-nml-tests

```diff
@@ -0,0 +1,276 @@
+        new_case_rest_n = run_cmd_assert_result(self, "./xmlquery REST_N --value", from_dir=case)
+        new_atm_freq    = run_cmd_assert_result(self, "./atmquery Frequency --value", from_dir=case)
+        self.assertEqual(new_case_rest_n, new_rest_n)
+        self.assertEqual(new_atm_freq, new_rest_n)
+
+    ###########################################################################
+    def test_atmchanges_are_preserved(self):
+    ###########################################################################
+        """
+        Test that atmchanges are not lost when eamxx setup is called
+        """
+        case = self._create_test("SMS.ne30_ne30.F2010-SCREAMv1 --no-build".split())
+
+        bm = run_cmd_assert_result(self, "./atmquery bm --value", from_dir=case)
+        new_bm = "42.42"
+        self.assertNotEqual(bm, new_bm)
+
+        run_cmd_assert_result(self, f"./atmchange bm={new_bm}", from_dir=case)
+        bm = run_cmd_assert_result(self, "./atmquery bm --value", from_dir=case)
+        self.assertEqual(bm, new_bm)
+
+        run_cmd_assert_result(self, "./case.setup", from_dir=case)
+
+        bm = run_cmd_assert_result(self, "./atmquery bm --value", from_dir=case)
+        self.assertEqual(bm, new_bm)
+
+    ###########################################################################
+    def test_manual_atmchanges_are_lost(self):
+    ###########################################################################
+        """
+        Test that manual atmchanges are lost when eamxx setup is called
+        """
+        case = self._create_test("SMS.ne30_ne30.F2010-SCREAMv1 --no-build".split())
+
+        orig_bm = run_cmd_assert_result(self, "./atmquery bm --value", from_dir=case)
+        new_bm = "42.42"
+        self.assertNotEqual(orig_bm, new_bm)
+
+        run_cmd_assert_result(self, f"./atmchange bm={new_bm} --no-buffer", from_dir=case)
+        bm = run_cmd_assert_result(self, "./atmquery bm --value", from_dir=case)
+        self.assertEqual(bm, new_bm)
+
+        run_cmd_assert_result(self, "./case.setup", from_dir=case)
+
+        bm = run_cmd_assert_result(self, "./atmquery bm --value", from_dir=case)
+        self.assertEqual(bm, orig_bm)
+
+    ###########################################################################
+    def test_manual_atmchanges_are_not_lost_hack_xml(self):
+    ###########################################################################
+        """
+        Test that manual atmchanges are not lost when eamxx setup is called if
+        xml hacking is enabled.
+        """
+        case = self._create_test("SMS.ne30_ne30.F2010-SCREAMv1 --no-build".split())
+
+        run_cmd_assert_result(self, f"./xmlchange SCREAM_HACK_XML=TRUE", from_dir=case)
+
+        bm = run_cmd_assert_result(self, "./atmquery bm --value", from_dir=case)
+        new_bm = "42.42"
+        self.assertNotEqual(bm, new_bm)
+
+        run_cmd_assert_result(self, f"./atmchange bm={new_bm} --no-buffer", from_dir=case)
```

Comment:
Do you think it  makes sense to allow `no-buffer` if the hack is on? Should we perhaps error out inside atmchange instead?

Reply:
If hack is on, then it doesn't matter if you buffer atmchanges or not, the namelists.xml file will not be automatically regenerated. I don't see `--no-buffer` having much use except to prevent the buffer-replay commands from themselves being buffered and for internal testing.

Reply:
Yeah, that's why I thought that specifying `--no-buffer` with the hack on might be suspicious, and we might want to error out. Perhaps we can issue a warning, in case the user forgot they're in hack mode?